alerta/plugins/heartbeat.py

```diff
@@ -19,7 +19,7 @@ def pre_receive(self, alert, **kwargs):
         HEARTBEAT_EVENTS = self.get_config('HEARTBEAT_EVENTS', default=['Heartbeat'], type=list, **kwargs)
 
         if alert.event in HEARTBEAT_EVENTS:
-            if 'timeout' in request.json:
+            if 'timeout' in request.json["commonLabels"]:
```

Comment:
Sorry but I don't see how this fixes the issue. Maybe I'm missing something. If so, please elaborate. Thanks.

Reply:
If there was a test that showed this worked after the change that would probably be enough for me to merge.

Reply:
yes, i tested on my productive service.
request.json is `{'receiver': 'alerta', 'status': 'firing', 'alerts': [{'status': 'firing', 'labels': {'alertname': 'DeadMansSwitch', 'collector': 'shm10_collector', 'event': 'Heartbeat', 'monitor': 'shm10_collector', 'severity': 'critical', 'timeout': '1800'}, 'annotations': {'description': 'This is a DeadMansSwitch meant to ensure that the entire Alerting pipeline is functional.', 'summary': 'Alerting DeadMansSwitch'}, 'startsAt': '2020-10-19T15:58:33.772089574Z', 'endsAt': '0001-01-01T00:00:00Z', 'generatorURL': 'http://bfdf51e9472c:9090/graph?g0.expr=vector%281%29&g0.tab=1'}], 'groupLabels': {'alertname': 'DeadMansSwitch'}, 'commonLabels': {'alertname': 'DeadMansSwitch', 'collector': 'shm10_collector', 'event': 'Heartbeat', 'monitor': 'shm10_collector', 'severity': 'critical', 'timeout': '1800'}, 'commonAnnotations': {'description': 'This is a DeadMansSwitch meant to ensure that the entire Alerting pipeline is functional.', 'summary': 'Alerting DeadMansSwitch'}, 'externalURL': 'http://42e370a7a6f5:9093', 'version': '4', 'groupKey': '{}:{alertname="DeadMansSwitch"}'}`
this dict not included timeout key. commonLabels field contain all keys

Reply:
This fix needs to go into the Prometheus webhook not the heartbeats plugin as it's specific to Prometheus Alertmanager.